@ViktorHofer I tried to run iOS and Android tests on this branch to make sure we don't regress them and they both failed to generate the app bundles.

### iOS
```
/Users/santifdezm/repos/runtime/artifacts/bin/System.Buffers.Tests/net5.0-Debug/ios-x64/AppBundle/modules.m:1:10: fatal error: 'mono/jit/jit.h' file not found
  #include <mono/jit/jit.h>
```

It seems like we're missing the include assets in the runtime pack. 

### Android
```
/Users/santifdezm/repos/runtime/eng/testing/tests.mobile.targets(36,5): error MSB4018: The "AndroidAppBuilderTask" task failed unexpectedly. [/Users/santifdezm/repos/runtime/src/libraries/System.Buffers/tests/System.Buffers.Tests.csproj]
/Users/santifdezm/repos/runtime/eng/testing/tests.mobile.targets(36,5): error MSB4018: System.ArgumentException: libmonosgen-2.0.a was not found in /Users/santifdezm/repos/runtime/artifacts/bin/System.Buffers.Tests/net5.0-Debug/android-x86/publish/ [/Users/santifdezm/repos/runtime/src/libraries/System.Buffers/tests/System.Buffers.Tests.csproj]
```

libmonosgen-2.0 is not in the runtime pack so it is missing on the publish dir. 

Marking as no merge in the meantime.

--- a/src/libraries/Directory.Build.targets
+++ b/src/libraries/Directory.Build.targets
@@ -103,7 +103,7 @@
     <!-- Corresponding libMono and system.native libraries are added to testhost shared framework -->
     <BinPlaceTargetFrameworks Include="$(NetCoreAppCurrent)-$(TargetOS)"
                               Condition="'$(BinPlaceTestSharedFramework)' == 'true'">
-      <RuntimePath>$(NETCoreAppTestSharedFrameworkPath)</RuntimePath>
+      <NativePath>$(NETCoreAppTestSharedFrameworkPath)</NativePath>
       <ItemName>TestHostBinPlaceItem</ItemName>
     </BinPlaceTargetFrameworks>
 

--- a/src/libraries/restore/runtime/runtime.depproj
+++ b/src/libraries/restore/runtime/runtime.depproj
@@ -4,6 +4,8 @@
     <NoWarn>$(NoWarn);NU1603;NU1605</NoWarn>
     <SwapNativeForIL Condition="'$(SwapNativeForIL)' == '' and ('$(Configuration)' == 'Debug' or '$(Coverage)' == 'true') and '$(RuntimeFlavor)' != 'Mono'">true</SwapNativeForIL>
     <BinPlaceForTargetVertical>false</BinPlaceForTargetVertical>
+    <BinPlaceNative>true</BinPlaceNative>
+    <BinPlaceRuntime>false</BinPlaceRuntime>
     <TargetFrameworks>netcoreapp3.0-Windows_NT;netcoreapp3.0-Unix;$(netcoreappCurrent)-Windows_NT;$(netcoreappCurrent)-Unix</TargetFrameworks>
   </PropertyGroup>
 
@@ -81,13 +83,13 @@
           Condition="'$(RuntimeFlavor)' == 'Mono'"
           AfterTargets="AfterResolveReferences">
     <ItemGroup>
-      <NativeBinPlaceItem Include="@(RuntimeFiles)" />
-      <NativeBinPlaceItem Include="@(MonoCrossFiles)">
+      <BinPlaceItem Include="@(RuntimeFiles)" />
+      <BinPlaceItem Include="@(MonoCrossFiles)">
         <DestinationSubDirectory>cross/</DestinationSubDirectory>
-      </NativeBinPlaceItem>
-      <NativeBinPlaceItem Include="@(MonoIncludeFiles)">
+      </BinPlaceItem>
+      <BinPlaceItem Include="@(MonoIncludeFiles)">
         <DestinationSubDirectory>include/%(RecursiveDir)</DestinationSubDirectory>
-      </NativeBinPlaceItem>
+      </BinPlaceItem>
     </ItemGroup>
     <ItemGroup Condition="'$(TargetsMobile)' != 'true'">
       <TestHostBinPlaceItem Include="@(RuntimeFiles)" />

